### PR TITLE
build Service Catalog and run tests

### PR DESCRIPTION
**Pull request by jboyd01**
initial stab at new job  to build  and test Service Catalog

working with @stevekuznetsov

### CODE REVIEW

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+parent: 'common/test_cases/origin.yml'
+overrides:
+  junit_analysis: False
+extensions:
+  sync_repos:
+    - name: "openshift-ansible"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: can remove this line

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    - type: "script"
+      title: "determine the release commit for origin images"
+      repository: "origin"
+      script: |-
+        jobs_repo="/data/src/github.com/openshift/aos-cd-jobs/"
+        git log -1 --pretty=%h > "${jobs_repo}/ORIGIN_COMMIT"
+        source hack/lib/init.sh
+        os::build::rpm::format_nvra > "${jobs_repo}/ORIGIN_BUILT_VERSION"
+    - type: "script"
+      title: "build an openshift-ansible release"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: can remove this task

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        hack/build-images.sh
+        sed -i 's|go/src|data/src|' _output/local/releases/rpms/origin-local-release.repo
+        sudo cp _output/local/releases/rpms/origin-local-release.repo /etc/yum.repos.d/
+    - type: "script"
+      title: "expose the kubeconfig"
+      script: |-
+        sudo chmod a+x /etc/ /etc/origin/ /etc/origin/master/
+        sudo chmod a+rw /etc/origin/master/admin.kubeconfig
+    - type: "script"
+      title: "ensure built version of origin is installed"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: can remove this task

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        fi
+        git checkout ${OPENSHIFT_ANSIBLE_TARGET_BRANCH}
+        hack/build-base-images.sh
+        OS_BUILD_ENV_PULL_IMAGE=true OS_BUILD_ENV_PRESERVE=_output/local hack/env OS_ONLY_BUILD_PLATFORMS='linux/amd64' hack/build-rpm-release.sh
+        sudo systemctl restart docker
+        hack/build-images.sh
+        sed -i 's|go/src|data/src|' _output/local/releases/rpms/origin-local-release.repo
+        sudo cp _output/local/releases/rpms/origin-local-release.repo /etc/yum.repos.d/
+    - type: "script"
+      title: "expose the kubeconfig"
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: should happen after `oc cluster up` and should point to correct kubeconfig -- @csrwng can you add the steps to write out the config?

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+      script: |-
+        oc cluster up --version=latest --service-catalog
+    - type: "script"
+      title: "Run Service Catalog E2E"
+      repository: "origin"
+      timeout: 1800
+      script: |-
+        oc login -u system:admin
+        SERVICECATALOGCONFIG=$KUBECONFIG cmd/service-catalog/go/src/github.com/kubernetes-incubator/service-catalog/bin/e2e.test  -v 10 -alsologtostderr 
+  generated_artifacts:
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: not necessary

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    origin_package_history.log: "sudo yum history info origin"
+    openshift_ansible_package_history.log: "sudo yum history info openshift-ansible"
+  system_journals:
```
Comment (stevekuznetsov) [inline sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: not useful

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml (hunk 1 above)
Comment (jboyd01) [reply sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: I removed only the line containing  - name: "openshift-ansible"

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml (hunk 6 above)
Comment (jboyd01) [reply sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: I removed generated_artifacts, but left the bit about system_journals:

**Comment by jboyd01** [comment]
Updated per Steve's initial review, I still need to work with @csrwng for the KUBECONFIG work

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml (hunk 4 above)
Comment (csrwng) [reply sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: Cluster up by default will place the admin kubeconfig at `/var/lib/origin/openshift.local.config/master/admin.kubeconfig`

File: sjb/config/test_cases/test_pull_request_origin_service_catalog.yml (hunk 6 above)
Comment (stevekuznetsov) [reply sjb/config/test_cases/test_pull_request_origin_service_catalog.yml]: Not sure `cluster up` will create any `systemd` services

**Comment by jboyd01** [comment]
@stevekuznetsov  can we merge this?  We have some outstanding enhancements to make which include 
1) formatting go unit test as junit output for better reporting
2) formatting ginko e2e as junit output output  for better reporting
3) execution of upstream integration tests
but for the initial delivery I'd like to call this complete.

**Comment by stevekuznetsov** [comment]
@jboyd01 awesome!

**Pull request merged**